cmd/ig/daemon.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	gadgetservice "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+func addDaemonCommand(rootCmd *cobra.Command, runtime runtime.Runtime) {
+	daemonCmd := &cobra.Command{
+		Use:   "daemon",
+		Short: "runs Inspektor Gadget as a daemon",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return nil
+		},
+	}
+
+	var socket string
+	var gid int
+
+	daemonCmd.PersistentFlags().IntVarP(
+		&gid,
+		"daemon-group-id",
+		"",
+		0,
+		"Group id that the unix socket should use (if daemon-socket is set to e.g. unix:///path/to.socket)")
+
+	daemonCmd.PersistentFlags().StringVarP(
+		&socket,
+		"daemon-socket",
+		"",
+		api.DefaultDaemonPath,
+		"The socket to listen on for new Inspektor Gadget requests. Can be a unix socket"+
+			" (unix:///path/to.socket) or a tcp socket (tcp://127.0.0.1:1234)")
+
+	daemonCmd.RunE = func(cmd *cobra.Command, args []string) error {
```

Comment:
if len(args) != 0, return an error?

Reply:
Good point, probably should use `Args:  cobra.NoArgs` instead of manual testing, though (although I only saw that used once in our codebase), right?

Reply:
There is one instance of each case:

```bash
$ git grep cobra.NoArgs
cmd/common/advise/seccomp-profile.go:           Args:         cobra.NoArgs,

$ git grep "len(args) != 0"
pkg/gadgets/run/tracer/run.go:  if len(args) != 0 {
```

 I think `cobra.NoArgs` is better.